lib/Onyx.js

```diff
@@ -1020,16 +1022,20 @@ function multiSet(data) {
  */
 function applyMerge(existingValue, changes) {
     const lastChange = _.last(changes);
+    if (_.isNull(lastChange)) {
+        return lastChange;
+    }
 
     if (_.isArray(existingValue) || _.isArray(lastChange)) {
         return lastChange;
     }
 
-    if (_.isObject(existingValue) || _.every(changes, _.isObject)) {
+    const changesToMake = _.filter(changes, change => Boolean(change));
```

Comment:
NAB - might be good to leave a comment here explaining the Boolean thing (i.e. I *think* I understand this - that it's filtering out any null/undefineds? - but I'm not 100% sure so others might not be either)